### PR TITLE
detect the terms of service app is requiring the user to sign them

### PR DESCRIPTION
**Pull request by mgallien**
will query the sign state for the terms of service and automatically pause or resume sync

### CODE REVIEW

**Review by claucambra — COMMENTED** [review]
I agree with Camila's suggestions, otherwise LGTM

File: src/gui/owncloudgui.cpp [hunk 1]
```diff
@@ -285,6 +285,16 @@ void ownCloudGui::slotTrayMessageIfServerUnsupported(Account *account)
     }
 }
 
+void ownCloudGui::slotNeedToAcceptTermsOfService(OCC::AccountPtr account)
+{
+    slotShowTrayMessage(
+        tr("Terms of service"),
+        tr("The account %1 is required you to acknowledge that you have read and agree to the terms of service of your server. "
+           "You will be redirected %2 to validate them.")
```
[the remark this answers is not in the document]
Comment (mgallien) [reply src/gui/owncloudgui.cpp:293]: done

File: src/gui/folder.cpp [hunk 2]
```diff
@@ -93,6 +93,16 @@ Folder::Folder(const FolderDefinition &definition,
...
         qCWarning(lcFolder, "Could not read system exclude file");
 
+    connect(_accountState.data(), &AccountState::termsOfServiceChanged,
+            this, [this] ()
+            {
+                if (_accountState->state() == AccountState::NeedToSignTermsOfService) {
+                    setSyncPaused(true);
+                } else {
+                    setSyncPaused(false);
+                }
```
[the remark this answers is not in the document]
Comment (mgallien) [reply src/gui/folder.cpp:103]: done

**Pull request merged**